types/src/crypto/asymmetric_key.rs

```diff
@@ -85,8 +81,10 @@ where
             return Err(Error::AsymmetricKey("too short".to_string()));
         }
 
-        let bytes = checksummed_hex::decode(&input)?;
-        let (tag, key_bytes) = bytes.split_at(1);
+        let (tag_bytes, key_bytes) = input.as_ref().split_at(2);
+
+        let tag = checksummed_hex::decode(&tag_bytes)?;
+        let key_bytes = checksummed_hex::decode(&key_bytes)?;
```

Comment:
the commit says "fix ..." - what was the problem with this implementation? I remember asking about it some time ago and was told it's equivalent but it's now being reverted.